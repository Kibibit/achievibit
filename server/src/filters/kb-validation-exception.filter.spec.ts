import { BadRequestException, HttpStatus } from '@nestjs/common';
import MockDate from 'mockdate';

import { hostMock, mockResponse } from '@kb-dev-tools';
import { KbValidationExceptionFilter } from '@kb-filters';

describe('KbValidationExceptionFilter', () => {
  it('should be defined', () => {
    expect(new KbValidationExceptionFilter()).toBeDefined();
  });

  it('should return pretty validation errors', async () => {
    MockDate.set('2000-05-04');
    const filter = new KbValidationExceptionFilter();
    const req = {
      path: '/mock-api-path',
      url: 'https://server.com/mock-api-path'
    };

    filter.catch(new BadRequestException(), hostMock(req, mockResponse));

    expect(mockResponse.status)
      .toHaveBeenCalledWith(HttpStatus.METHOD_NOT_ALLOWED);
    expect(mockResponse.json).toHaveBeenCalledTimes(1);
    expect(mockResponse.json.mock.calls[0][0]).toMatchSnapshot();
  });
});
