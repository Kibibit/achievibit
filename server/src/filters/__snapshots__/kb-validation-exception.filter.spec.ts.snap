// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KbValidationExceptionFilter should return pretty validation errors 1`] = `
PublicError {
  "error": "Bad Request",
  "name": "BadRequestException",
  "path": "https://server.com/mock-api-path",
  "statusCode": 405,
  "timestamp": "2000-05-04T00:00:00.000Z",
}
`;
